### PR TITLE
U049, U052: The player can use Spirit Points to purchase Life Upgrade, The Player can view a list of all players

### PR DESCRIPTION
**Pull request by enlne**
displaying the last seen date for the all player list isn't formatted

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -221,6 +228,45 @@ io.on('connection', function(socket) {
     }
   });
 
+  socket.on('cl_buy_life', async (done) => {
```
Comment (Kailang) [inline src/index.js]: Change 'cl_buy_life' to 'cl_store_buy_life'.

File: src/index.js [hunk 2]
```diff
@@ -221,6 +228,45 @@ io.on('connection', function(socket) {
...
+        magicPointsCount: user.magicPointsCount,
+        lifeUpgrade: user.lifeUpgrade,
+      });
+      return done(success({spiritPointsCount: user.spiritPointsCount, lifeUpgrade: user.lifeUpgrade}));
+    } catch (e) {
+      return done(error('purchase failed'));
+    }
+  });
+
+  socket.on('cl_all_players', async (done) => {
```
Comment (Kailang) [inline src/index.js]: Change 'cl_all_players' to 'cl_get_players'.

File: src/index.js (hunk 1 above)
Comment (enlne) [reply src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (enlne) [reply src/index.js]: done

**Comment by enlne** [comment]
refactor to React complete

File: src/index.js [hunk 3]
```diff
@@ -221,6 +231,45 @@ io.on('connection', function(socket) {
...
   });
 
+  socket.on('cl_store_buy_life', async (done) => {
+    debug('cl_store_buy_life');
+
+    if (!user) return done(error('forbidden'));
+    user.spiritPointsCount -= LIFE_UPGRADE_COST;
+    user.lifeUpgrade += LIFE_UPGRADE_AMOUNT;
+    try {
+      await User.findByIdAndUpdate(user.id, user);
```
Comment (Kailang) [inline src/index.js]: Do not use `user` directly in findByIdAndUpdate.

File: src/index.js [hunk 4]
```diff
@@ -221,6 +231,45 @@ io.on('connection', function(socket) {
...
+      socket.emit(SV_REFRESH_USER, {
```
Comment (Kailang) [inline src/index.js]: No need for this event anymore.

**Pull request merged**